confluence_utils/markdown_file.py

```diff
@@ -78,7 +33,97 @@ def render_confluence_content(self) -> Tuple[str, List[str]]:
                 DirectiveConfluenceToc(),
             ],
         )
-        body = markdown(self.markdown_content)
-        attachments: List[str] = []
 
-        return body, attachments
+        self.html = md(self._raw_markdown)
+        self._attachments = renderer.attachments
+
+    @property
+    def title(self) -> Optional[str]:
+        return self._get_attribute("title")
+
+    @title.setter
+    def title(self, title: str) -> None:
+        self._set_attribute("title", title)
+
+    @property
+    def attachments(self) -> List[str]:
+        return [
+            os.path.normpath(
+                os.path.join(os.path.dirname(self.path), attachment)
+            )
+            for attachment in self._attachments
+        ]
+
+    def _get_attribute(self, attribute_name: str) -> Optional[str]:
+        return self._front_matter.get(attribute_name, None)
+
+    def _set_attribute(self, attribute_name: str, value: str) -> None:
+        self._front_matter[attribute_name] = value
+        self._save()
+
+    def _get_attribute_for_space(
+        self, attribute_name: str, space_key: str
+    ) -> Optional[str]:
+        return (
+            self._front_matter.get("spaces", {})
+            .get(space_key, {})
+            .get(attribute_name, None)
+        )
+
+    def _set_attribute_for_space(
+        self, attribute_name: str, space_key: str, value: str
+    ) -> None:
+        spaces = self._front_matter.get("spaces", {})
+        space = spaces.get(space_key, {})
+        space[attribute_name] = value
+        spaces[space_key] = space
+        self._front_matter["spaces"] = spaces
+        self._save()
+
+    def get_page_id_for_space(self, space_key: str) -> Optional[str]:
+        return self._get_attribute_for_space("page_id", space_key)
+
+    def set_page_id_for_space(self, page_id: str, space_key: str) -> None:
+        self._set_attribute_for_space("page_id", space_key, page_id)
+
+    def get_parent_page_id_for_space(self, space_key: str) -> Optional[str]:
+        return self._get_attribute_for_space("parent_page_id", space_key)
+
+    def set_parent_page_id_for_space(
+        self, parent_page_id: str, space_key: str
+    ) -> None:
+        self._set_attribute_for_space(
+            "parent_page_id", space_key, parent_page_id
+        )
+
+    @property
+    def parent_file_path(self) -> Optional[str]:
+        parent_file_path = self._get_attribute("parent_file_path")
+        return (
+            os.path.normpath(
+                os.path.join(
+                    os.path.dirname(self.path),
+                    parent_file_path,
+                )
+            )
+            if parent_file_path is not None
+            else None
+        )
```

Comment:
should this error if a path is given but not findable?

Reply:
@ttrout55 the parse function will throw a file not found error if the path is given but the file doesn't exist.